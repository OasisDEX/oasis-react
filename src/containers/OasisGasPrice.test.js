/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { fromJS } from 'immutable';

import {
  OasisGasPriceWrapper,
  mapStateToProps,
  mapDispatchToProps
} from './OasisGasPrice';
import { shallow } from 'enzyme';

describe('(Container) OasisGasPrice', () => {
  const state = fromJS(global.storeMock);
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });


  it('will receive right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(
      <OasisGasPriceWrapper {...props}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
