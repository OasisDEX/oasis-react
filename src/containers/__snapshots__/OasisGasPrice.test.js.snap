// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) OasisGasPrice should render 1`] = `
<div
  style={
    Object {
      "background": "#fff",
      "padding": "4px 7px",
    }
  }
>
  <div />
</div>
`;

exports[`(Container) OasisGasPrice will receive right actions 1`] = `
Object {
  "actions": Object {},
}
`;

exports[`(Container) OasisGasPrice will receive right props 1`] = `
Object {
  "currentGasPrice": "30000000000",
  "latestBlockNumber": 6716998,
  "latestEthereumPrice": Immutable.Map {
    percent_change_7d: "-16.34",
    available_supply: "98614682.0",
    max_supply: null,
    total_supply: "98614682.0",
    name: "Ethereum",
    rank: "2",
    percent_change_1h: "-0.99",
    price_usd: "377.388",
    percent_change_24h: "-8.14",
    last_updated: "1522866553",
    market_cap_usd: "37215997787.0",
    symbol: "ETH",
    24h_volume_usd: "1329870000.0",
    id: "ethereum",
    price_btc: "0.0557221",
},
}
`;
